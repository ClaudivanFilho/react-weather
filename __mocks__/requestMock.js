export function fetchRequestMock() {
  return new Promise((resolve, reject) => {
    process.nextTick(
      resolve('success')
    );
  });
}