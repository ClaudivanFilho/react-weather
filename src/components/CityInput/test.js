import React from 'react';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

import { CityInput } from './index';
import { setTimeout } from 'timers';
import { fetchRequestMock } from '../../../__mocks__/requestMock' 

configure({ adapter: new Adapter() });

test('Test CityInput Component - simulate onChange input event', (done) => {
  const component = shallow(<CityInput fetchCurrentWeather={fetchRequestMock} fetchForecast={fetchRequestMock}/>);
  expect(component.find('input').props().value).toEqual(undefined);
  component.find('input').simulate('change', {target: {value: 'Campina Grande'}, persist: () => {}});
  // It's necessary because of debounce funcion presents in the input onChange handle
  setTimeout(() => {
    done();
  }, 2000)
});

test('Test CityInput Snapshot', () => {
  const component = renderer.create(
    <CityInput />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})