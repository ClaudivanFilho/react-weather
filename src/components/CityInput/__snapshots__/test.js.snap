// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CityInput Snapshot 1`] = `
<div
  className="ui form text-right"
>
  <div
    className="inline fields float-right"
  >
    <label>
       Move the marker to a location or type a city:  
    </label>
    <input
      className="col-sm-4"
      onChange={[Function]}
      type="text"
    />
  </div>
</div>
`;
